fastflix/widgets/panels/status_panel.py

```diff
@@ -111,6 +121,40 @@ def update_bitrate(self, bitrate):
     def update_title_bar(self):
         pass
 
+    def set_started_at(self, msg):
+        try:
+            started_at = datetime.datetime.fromisoformat(msg.split("__")[-1])
+        except Exception as e:
+            logger.warning(f"Unable to parse start time - {str(e)}")
+            return
+
+        self.started_at = started_at
+
+    def update_time_elapsed(self):
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        if not self.started_at:
+            logger.warning("Unable to update time elapsed because start time isn't set")
+            return
+
+        try:
+            time_elapsed = now - self.started_at
+        except Exception as e:
+            logger.warning(f"Unable to calculate elapsed time - {str(e)}")
+            return
+
+        self.time_elapsed_label.setText(f"{t('Time Elapsed')}: {timedelta_to_str(time_elapsed)}")
+
+    def on_status_update(self, msg):
+        update_type = msg.split("__")[0]
+
+        if update_type in ["complete", "cancelled", "error"]:
+            self.time_elapsed_ticker_thread.running = False
+            self.time_elapsed_ticker_thread = ElapsedTimeTicker(self)
```

Comment:
As we can just run this in the background the entire time the app is running, I would not bother re-declaring this, and instead just put it's cleanup (sending the signal to exit) in the `closeEvent` function (a PyQT thing).

```python
    def closeEvent(self, event):
        self.new_properly_named_signal_to_stop_thread.emit() # Or just .terminate the thread, don't need to be fancy now that I'm thinking about it more
        super().closeEvent(event)
```

Instead I would just set `started_at` to None, and make that the first check in `update_time_elapsed` (and remove warning)  so that it stops updating when done.

Reply:
Okay! I thought we could discuss the whole thread communicating thing in this thread. I've made an attempt to rewrite it to only communicate via signals in c1e86d2.

Instead of setting `started_at` to `None` when status completed/cancelled/error is received like you suggested I made the ticker thread itself listen to the status update signals as well and stop/start emitting the tick signal based on status. I realise that might be a bit wonky, but it *seems* to work well and it feels more elegant to me. Would love to hear your thoughts and if there's anything else I've missed! 🙂

P.S.: I also found the `cleanup` function in the `status_panel` widget itself. Which is why I didn't use `closeEvent`. I couldn't get `closeEvent` to be called. Might be because it happens somewhere else? But `cleanup` is run and the loop exits on close.

Reply:
This is great! This is now better than I would have wrote it, so thank you so much for making it so well!